vtex-apps/login
# Feature/user identifier extension

#### What is the purpose of this pull request?
Allow identifiers other than `email` to be configured when logging in with password to the store.

#### What problem is this solving?
This features lets the enterprise accounts create their own extension app to change the login with password. They will add their own component to replace the `email` Input in the login form to accept other identifiers, such as a National Identity Document. It only works if the identifier can effectively be transformed into an email. More details in the updated documentation `README.md`.

This also adds some new messages to the Store Front configurations, which can be edited by the site-editor.

#### How should this be manually tested?

Go into
https://rafaprtest--storecomponents.myvtex.com/
You can test two things:
- If the messages changed in the site editor work correctly. There are prints below to show it.
- The Identifier Extension app works correctly. I created an example which takes a string and resolves it to an email by adding `@gmail.com` to the end of it. You can test with the following steps:
  - Create a store account with a `gmail` email and set a password (eg. `example@gmail.com`).
  - Then, try to log in writing only what precedes the `@gmail.com` (eg. write only `example`). The extension app will automatically add `@gmail.com` and you should log in normally!

#### Screenshots or example usage

![image](https://user-images.githubusercontent.com/22064061/65471184-a6daa480-de44-11e9-8d5a-dc2511efa2a7.png)

![image](https://user-images.githubusercontent.com/22064061/65471199-b2c66680-de44-11e9-89a1-1d2125818a97.png)

![image](https://user-images.githubusercontent.com/22064061/65471213-be199200-de44-11e9-8af7-a6958f3fd76f.png)

#### Types of changes

#### Clubhouse stories
[ch20521]
